SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -77,35 +77,73 @@ class const_ref {
+                                            out);
   }
 
 private:
   template <typename DataT, int NumElems>
-  static simd<DataT, NumElems>
-  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref) {
-    return simd_by_const_ref;
+  static void
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref,
+                         DataT *out) {
+    simd_by_const_ref.copy_to(out);
   }
 };
 
-int main(int argc, char **argv) {
-  sycl::queue queue{esimd_test::ESIMDSelector{},
-                    esimd_test::createExceptionHandler()};
+// Struct that calls simd in provided context and then verifies obtained result.
+struct Verifier {
+  template <typename DataT, int NumElems, typename TestCaseT>
+  static bool verify(sycl::queue &queue, const std::vector<DataT> ref_data,
+                     const std::string &data_type) {
+    bool passed = true;
+
+    shared_vector<DataT> result(NumElems, shared_allocator<DataT>{queue});
+    shared_vector<DataT> shared_ref_data(ref_data.begin(), ref_data.end(),
+                                         shared_allocator<DataT>{queue});
+
+    queue.submit([&](sycl::handler &cgh) {
+      const auto ref = shared_ref_data.data();
+      auto out = result.data();
+
+      cgh.single_task<Kernel<DataT, NumElems, TestCaseT>>(
+          [=]() SYCL_ESIMD_KERNEL {
+            TestCaseT::template call_simd_ctor<DataT, NumElems>(ref, out);
+          });
+    });
+    queue.wait_and_throw();
+
+    for (size_t i = 0; i < result.size(); ++i) {
+      if (!are_bitwise_equal(ref_data[i], result[i])) {
+        passed = false;
+
+        auto test_description = TestDescription<DataT, NumElems>(
+            TestCaseT::get_description(), data_type, result[i], ref_data[i], i);
+        log::fail(test_description);
+      }
+    }
+
+    return passed;
+  }
+};
 
-  bool passed{true};
+int main(int argc, char **argv) {
+  sycl::queue q{esimd_test::ESIMDSelector{},
+                esimd_test::createExceptionHandler()};
+  bool pass = true;
 
   const auto types{get_tested_types<tested_types::all>()};
   const auto dims{get_all_dimensions()};
 
-  passed &= for_all_types_and_dims<test, initializer>(types, dims, queue);
-  passed &= for_all_types_and_dims<test, var_decl>(types, dims, queue);
-  passed &= for_all_types_and_dims<test, rval_in_expr>(types, dims, queue);
-  passed &= for_all_types_and_dims<test, const_ref>(types, dims, queue);
+  pass &=
+      for_all_types_and_dims<run_test, initializer, Verifier>(types, dims, q);
+  pass &= for_all_types_and_dims<run_test, var_decl, Verifier>(types, dims, q);
+  pass &=
+      for_all_types_and_dims<run_test, rval_in_expr, Verifier>(types, dims, q);
+  pass &= for_all_types_and_dims<run_test, const_ref, Verifier>(types, dims, q);
 
-  std::cout << (passed ? "=== Test passed\n" : "=== Test FAILED\n");
-  return passed ? 0 : 1;
+  std::cout << (pass ? "=== Test passed\n" : "=== Test FAILED\n");
+  return pass ? 0 : 1;
```

Comment:
```suggestion
  return ok ? 0 : 1;
```
Just as a suggestion, if you are looking for a shorter naming to use over the tests. Anyway, just be consistent with it - that's the same flag on the all levels anyway. So if you change it somewhere - let's change it everywhere.
Actually, the `passed` looks good to me, as it's hard to misread it as a verb instead of noun/adj.

Reply:
I make a better solution, I've done it in this commit: https://github.com/intel/llvm-test-suite/pull/614/commits/f67072db7233c3435d7b090999e75f79c333019b

Reply:
OK, so we stick with the `passed` then :)